security/pkg/server/ca/monitoring.go

```diff
@@ -60,11 +60,21 @@ var (
 		"The unix timestamp, in seconds, when Citadel root cert will expire. "+
 			"A negative time indicates the cert is expired.",
 	)
+	rootCertExpirySeconds = monitoring.NewGauge(
```

Comment:
To me thes metrics are useless, because we donot update it after startup.  I would prefer recording the expire time instead. Or we should update the time to expire periodically

Reply:
Great catch! The expiry timestamp is already recorded. Since this is a user request #51037, I'd like to update the time to expire periodically similar to what we do with `cert_expiry_seconds` or `istiod_uptime_seconds`.

I believe switching to the [`NewDerivedGauge()` ](https://github.com/istio/istio/blob/9bf6c0d1214a1b0a40a1f4596e16a87ddd7c7d1a/pkg/monitoring/monitoring.go#L209) will accomplish this.

Reply:
,make sense to me

Reply:
PTAL, I have updated.